ipamd/ipamd.go

```diff
@@ -1000,17 +997,18 @@ func (c *IPAMContext) nodeIPPoolReconcile(interval time.Duration) {
 }
 
 func (c *IPAMContext) eniIPPoolReconcile(ipPool map[string]*datastore.AddressInfo, attachedENI awsutils.ENIMetadata, eni string) {
-	for _, localIP := range attachedENI.LocalIPv4s {
-		if localIP == c.primaryIP[eni] {
-			log.Debugf("Reconcile and skip primary IP %s on ENI %s", localIP, eni)
+	for _, privateIPv4 := range attachedENI.IPv4Addresses {
+		strPrivateIPv4 := aws.StringValue(privateIPv4.PrivateIpAddress)
+		if strPrivateIPv4 == c.primaryIP[eni] {
+			log.Debugf("Reconcile and skip primary IP %s on ENI %s", strPrivateIPv4, eni)
 			continue
 		}
```

Comment:
Could this check be this instead?
```go
if aws.BoolValue(privateIPv4.Primary) {
	log.Debugf("Skip primary IP %s on ENI %s during reconcile", strPrivateIPv4, eni)
	continue
}
```

Reply:
yes, I can change it to that in a followup. ++